addons/mrp_subcontracting/models/stock_picking.py

```diff
@@ -70,6 +70,8 @@ def _action_done(self):
             amounts = [move_line.qty_done for move_line in move.move_line_ids]
             len_amounts = len(amounts)
             productions = production._split_productions({production: amounts}, set_consumed_qty=True)
+            for production in productions:
+                production.move_finished_ids.move_line_ids.write({'qty_done': 0})
```

Comment:
sometimes the `move_finished_ids` are set to done/cancel with some bad custo, if it's the case here it will throw an error as the move state is wrong.

Reply:
I'm not sure we should handle bad custo in standard. But this fix in subcontracting is not the best, I had to make it because if the quantity is set on the move_finished_ids, the _post_inventory won't write the lot_id on the stock.move.line and it will raise a Usererror. I could do a more complete fix but since it's stable I would like to avoid doing too much change.